app/modules/integrations/integrations_service.py

```
@@ -1636,3 +1797,530 @@ async def save_integration(
+                        new_tokens["access_token"]
+                    )
+                    if new_tokens.get("refresh_token"):
+                        auth_data["refresh_token"] = encrypt_token(
+                            new_tokens["refresh_token"]
+                        )
+                    auth_data["expires_at"] = datetime.fromtimestamp(
+                        new_tokens["expires_at"], tz=timezone.utc
+                    ).isoformat()
+
+                    setattr(db_integration, "auth_data", auth_data)
+                    setattr(db_integration, "updated_at", datetime.utcnow())
```

Ruff (0.14.3)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)